### ISSUE TITLE
SGD viewer issue

### ISSUE THREAD

**Issue by azazellochg** [scipion-em/scipion-em-relion#20]
When clicking on Changes in offsets and angles or Show average Pmax, I get:
```
 Computing average changes in offset, angles, and class membership
Computing data for iteration; 001
Exception in Tkinter callback
Traceback (most recent call last):
  File "/lmb/home/gsharov/soft/scipion-devel/software/lib/python2.7/lib-tk/Tkinter.py", line 1541, in __call__
    return self.func(*args)
  File "/lmb/home/gsharov/soft/scipion-devel/pyworkflow/gui/form.py", line 972, in _visualizeVar
    self.visualizeCallback(self.paramName)
  File "/lmb/home/gsharov/soft/scipion-devel/pyworkflow/viewer.py", line 314, in _visualizeParam
    views = self._getVisualizeDict()[paramName](paramName)
  File "/lmb/home/gsharov/soft/scipion-em-plugins/scipion-em-relion/relion/viewers.py", line 472, in _showChanges
    mdOptimiser = md.RowMetaData(fn)
  File "/lmb/home/gsharov/soft/scipion-devel/pyworkflow/em/metadata/classes.py", line 176, in __init__
    self.read(filename)
  File "/lmb/home/gsharov/soft/scipion-devel/pyworkflow/em/metadata/classes.py", line 188, in read
    self._md.read(filename)
XmippError: FileName::isMetaData: File: 'Runs/001688_ProtRelionInitialModel/extra/relion_it001_optimiser.star' does not exist
```
This is because output star is written every 10 iterations (depends on settings)

**Comment by azazellochg** [comment]
@delarosatrevin any chance you had a look at this?

**Comment by delarosatrevin** [comment]
Fixed in 4785438dd73d. 

I have created a nice decorator "@protected_show" that can be used to wrap show functions and if exceptions are raised, it will be shown as an error message.
